### PR TITLE
@alloy => Add support for sharing web view caches

### PR DESCRIPTION
**Pull request by orta**
TODO:
- [x] Add `/dev/blank` to force https://github.com/artsy/force/issues/3673
- [x] Add `/dev/blank` to martsy https://github.com/artsy/microgravity/pull/1138 

Does some more aggressive caching in our WKWebViews.

### CODE REVIEW

File: Artsy.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -4978,7 +4992,7 @@
 					"-DAR_SHOW_ALL_DEBUG=1",
 					"-Werror=nonnull",
 				);
-				PROVISIONING_PROFILE = "d3a0cd2c-e754-449c-80a5-285c4bb5f504";
+				PROVISIONING_PROFILE = "09dca157-822f-4776-b30f-45f5f49c2810";
```
Comment (orta) [inline Artsy.xcodeproj/project.pbxproj]: screen shot 2015-10-26 at 12 20 40

**Comment by orta** [comment]
OK, with optimal WIFI at home:

```
Without:

2015-10-27 10:42:56.361 Artsy[47468:1544168] Started request: http://m-staging.artsy.net/auction/input-output 
2015-10-27 10:42:56.362 Artsy[47468:1544168] Started request: - 0.000755012035369873
2015-10-27 10:43:00.112 Artsy[47468:1544168] Now presenting the webview - 3.750616014003754
2015-10-27 10:43:00.145 Artsy[47468:1544168] Totally finished showing the webview - 3.78433096408844

2015-10-27 10:43:21.151 Artsy[47468:1544168] Started request: http://m-staging.artsy.net/how-auctions-work 
2015-10-27 10:43:21.152 Artsy[47468:1544168] Started request: - 0.000487983226776123
2015-10-27 10:43:23.456 Artsy[47468:1544168] Now presenting the webview - 2.304744005203247
2015-10-27 10:43:23.818 Artsy[47468:1544168] Totally finished showing the webview - 2.666263997554779


2015-10-27 10:44:04.080 Artsy[47468:1544168] Started request: http://m-staging.artsy.net/auction/land-benefit-auction-2015 - 
2015-10-27 10:44:04.080 Artsy[47468:1544168] Started request: %@ - 0.0002840161323547363
2015-10-27 10:44:06.888 Artsy[47468:1544168] Now presenting the webview - 2.808113992214203
2015-10-27 10:44:07.323 Artsy[47468:1544168] Totally finished showing the webview - 3.243281006813049

2015-10-27 10:44:29.324 Artsy[47468:1544168] Started request: http://m-staging.artsy.net/galerie-deste - 
2015-10-27 10:44:30.979 Artsy[47468:1544168] Now presenting the webview - 1.655076026916504
2015-10-27 10:44:31.350 Artsy[47468:1544168] Totally finished showing the webview - 2.026199996471405

[ slighly pre-cached, I ran through these on another boot]

2015-10-27 10:44:59.478 Artsy[47468:1544168] Started request: http://m-staging.artsy.net/galerie-deste/shows
2015-10-27 10:44:59.478 Artsy[47468:1544168] Started request: %@ - 0.000299990177154541
2015-10-27 10:45:00.921 Artsy[47468:1544168] Now presenting the webview - 1.443224012851715
2015-10-27 10:45:01.063 Artsy[47468:1544168] Totally finished showing the webview - 1.585092008113861

2015-10-27 10:45:21.377 Artsy[47468:1544168] Started request: http://m-staging.artsy.net/galerie-deste/artists - 1.966953277587891e-06
2015-10-27 10:45:21.377 Artsy[47468:1544168] Started request: %@ - 0.0002709627151489258
2015-10-27 10:45:22.937 Artsy[47468:1544168] Now presenting the webview - 1.5607990026474
2015-10-27 10:45:23.234 Artsy[47468:1544168] Totally finished showing the webview - 1.857128977775574

2015-10-27 10:45:59.349 Artsy[47468:1544168] Started request: http://m-staging.artsy.net/galerie-deste/artist/joan-miro - 9.5367431640625e-07
2015-10-27 10:45:59.350 Artsy[47468:1544168] Started request: %@ - 0.000265955924987793
2015-10-27 10:46:00.610 Artsy[47468:1544168] Now presenting the webview - 1.260355949401855
2015-10-27 10:46:01.153 Artsy[47468:1544168] Totally finished showing the webview - 1.803614974021912

--- AFTER


2015-10-27 11:39:42.695 Artsy[49104:1617154] Started a precache
2015-10-27 11:39:42.696 Artsy[49104:1617154] Started a precache
2015-10-27 11:39:42.707 Artsy[49104:1617154] Started a precache
2015-10-27 11:39:43.716 Artsy[49104:1617154] Finished a precache
2015-10-27 11:39:43.853 Artsy[49104:1617154] Finished a precache
2015-10-27 11:39:43.968 Artsy[49104:1617154] Finished a precache

2015-10-27 11:39:48.481 Artsy[49104:1617154] Started request: http://m-staging.artsy.net/auction/land-benefit-auction-2015 - 7.987022399902344e-06
2015-10-27 11:39:48.630 Artsy[49104:1617154] Started a precache
2015-10-27 11:39:49.568 Artsy[49104:1617154] Finished a precache
2015-10-27 11:39:50.032 Artsy[49104:1617154] Now presenting the webview - 1.550096988677979
2015-10-27 11:39:50.479 Artsy[49104:1617154] Totally finished showing the webview - 1.997285008430481

2015-10-27 11:39:52.399 Artsy[49104:1617154] Started request: http://m-staging.artsy.net/how-auctions-work - 1.001358032226562e-05
2015-10-27 11:39:52.546 Artsy[49104:1617154] Started a precache
2015-10-27 11:39:53.079 Artsy[49104:1617154] Now presenting the webview - 0.6798539757728577
2015-10-27 11:39:53.232 Artsy[49104:1617154] Totally finished showing the webview - 0.8326430320739746
2015-10-27 11:39:53.320 Artsy[49104:1617154] Finished a precache

2015-10-27 11:40:03.454 Artsy[49104:1617154] Started request: http://m-staging.artsy.net/galerie-deste - 1.966953277587891e-06
2015-10-27 11:40:03.603 Artsy[49104:1617154] Started a precache
2015-10-27 11:40:03.626 Artsy[49104:1617154] Application tried to push a nil view controller on target <ARNavigationController: 0x7fcf998e7c00>.
2015-10-27 11:40:04.525 Artsy[49104:1617154] Finished a precache
2015-10-27 11:40:04.820 Artsy[49104:1617154] Now presenting the webview - 1.366243004798889
2015-10-27 11:40:05.178 Artsy[49104:1617154] Totally finished showing the webview - 1.724057972431183

2015-10-27 11:40:07.620 Artsy[49104:1617154] Started request: http://m-staging.artsy.net/galerie-deste/shows - 2.026557922363281e-06
2015-10-27 11:40:07.778 Artsy[49104:1617154] Started a precache
2015-10-27 11:40:08.179 Artsy[49104:1617154] Totally finished showing the webview - 0.5588940382003784
2015-10-27 11:40:08.179 Artsy[49104:1617154] Now presenting the webview - 0.5593170523643494
2015-10-27 11:40:08.410 Artsy[49104:1617154] Finished a precache

2015-10-27 11:40:17.110 Artsy[49104:1617154] Started request: http://m-staging.artsy.net/galerie-deste/artists - 3.039836883544922e-06
2015-10-27 11:40:17.270 Artsy[49104:1617154] Started a precache
2015-10-27 11:40:17.953 Artsy[49104:1617154] Totally finished showing the webview - 0.8433000445365906
2015-10-27 11:40:17.953 Artsy[49104:1617154] Now presenting the webview - 0.843673050403595
2015-10-27 11:40:18.007 Artsy[49104:1617154] Finished a precache

2015-10-27 11:40:21.059 Artsy[49104:1617154] Started request: http://m-staging.artsy.net/galerie-deste/artist/joan-miro - 3.993511199951172e-06
2015-10-27 11:40:21.220 Artsy[49104:1617154] Started a precache
2015-10-27 11:40:21.744 Artsy[49104:1617154] Now presenting the webview - 0.6850460171699524
2015-10-27 11:40:21.843 Artsy[49104:1617154] Finished a precache
2015-10-27 11:40:21.900 Artsy[49104:1617154] Totally finished showing the webview - 0.8406599760055542
```

**Comment by orta** [comment]
OK with the network link conditioner on 3G:

No pre-warmed cache
```
 Started request: http://m-staging.artsy.net/auction/land-benefit-auction-2015 - 
 Now presenting the webview - 6.491276979446411
 Totally finished showing the webview - 8.076353967189789

 Started request: http://m-staging.artsy.net/how-auctions-work -
 Now presenting the webview - 3.377477049827576
 Totally finished showing the webview - 4.249734997749329

 Started request: http://m-staging.artsy.net/auction/input-output - 
 Now presenting the webview - 4.07845801115036
 Totally finished showing the webview - 5.413411021232605


 Started request: http://m-staging.artsy.net/galerie-deste - 
 Totally finished showing the webview - 4.631062984466553
 Now presenting the webview - 4.631249010562897

 Started request: http://m-staging.artsy.net/galerie-deste/shows - 
 Totally finished showing the webview - 4.07040399312973
 Now presenting the webview - 4.070629000663757

 Started request: http://m-staging.artsy.net/galerie-deste/artists -
 Totally finished showing the webview - 4.64664900302887
 Now presenting the webview - 4.646857023239136

 Started request: http://m-staging.artsy.net/galerie-deste/artist/joan-miro - 
 Totally finished showing the webview - 11.47880202531815
 Now presenting the webview - 11.47903203964233
```

With cache
```

 Started request: http://m-staging.artsy.net/auction/land-benefit-auction-2015
 Started a precache
 Now presenting the webview - 4.104530990123749
 Finished a precache
 Totally finished showing the webview - 5.409255027770996

 Started request: http://m-staging.artsy.net/how-auctions-work 
 Started a precache
 Totally finished showing the webview - 4.01667594909668
 Now presenting the webview - 4.01691597700119
 Finished a precache


 Started request: http://m-staging.artsy.net/auction/input-output 
 Started a precache
 Finished a precache
 Now presenting the webview - 4.625159978866577
 Totally finished showing the webview - 5.355714023113251

 Started request: http://m-staging.artsy.net/galerie-deste
 Started a precache
 Now presenting the webview - 3.428629994392395
 Totally finished showing the webview - 3.719372987747192
 Finished a precache

 Started request: http://m-staging.artsy.net/galerie-deste/shows
 Started a precache
 Totally finished showing the webview - 1.448808968067169
 Now presenting the webview - 1.449114978313446
 Finished a precache

 Started request: http://m-staging.artsy.net/galerie-deste/artists
 Started a precache
 Totally finished showing the webview - 1.556010961532593
 Now presenting the webview - 1.55623596906662
 Finished a precache

 Started request: http://m-staging.artsy.net/galerie-deste/artist/joan-miro
 Started a precache
 Now presenting the webview - 1.143483996391296
 Finished a precache
 Totally finished showing the webview - 1.570091962814331

```

**Comment by orta** [comment]
Both of the above examples had the exact same potential for system-wide pre-caching - I never reinstalled, or reset anything in-between builds and have been doing all day. If the OS was caching, we were getting it for both.

**Comment by alloy** [comment]
Tests pass, but needs a rebase :-/

**Comment by orta** [comment]
Pushed rebase, gonna make a force PR now

**Comment by orta** [comment]
Alright, all dependant PRs are merged :+1:

**Comment by alloy** [comment]
Sweet :+1:

**Pull request merged**